Content.Goobstation.Server/Xenobiology/XenobiologySystem.Breeding.cs

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: 2025 August Eymann <august.eymann@gmail.com>
+// SPDX-FileCopyrightText: 2025 GoobBot <uristmchands@proton.me>
+// SPDX-FileCopyrightText: 2025 SolsticeOfTheWinter <solsticeofthewinter@gmail.com>
+// SPDX-FileCopyrightText: 2025 TheBorzoiMustConsume <197824988+TheBorzoiMustConsume@users.noreply.github.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+using Content.Goobstation.Shared.Xenobiology;
+using Content.Goobstation.Shared.Xenobiology.Components;
+using Content.Shared.Nutrition.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Goobstation.Server.Xenobiology;
+
+/// <summary>
+/// This handles slime breeding and mutation.
+/// </summary>
+public partial class XenobiologySystem
+{
+    private readonly TimeSpan _updateInterval = TimeSpan.FromSeconds(1);
+    private TimeSpan _nextUpdateTime;
+
+    private void InitializeBreeding()
+    {
+        _nextUpdateTime = _gameTiming.CurTime + _updateInterval;
+    }
+
+    // Mitosis doesn't need to be checked every frame.
+    private void UpdateBreeding()
+    {
+        if (_nextUpdateTime > _gameTiming.CurTime)
+            return;
+
+        _nextUpdateTime = _gameTiming.CurTime + _updateInterval;
+        UpdateMitosis();
+    }
+
+    // Checks slime entity hunger threshholds, if the threshhold required by SlimeComponent is met -> DoMitosis.
+    private void UpdateMitosis()
+    {
+        var eligibleSlimes = new HashSet<Entity<SlimeComponent, MobGrowthComponent, HungerComponent>>();
+
+        var query = EntityQueryEnumerator<SlimeComponent, MobGrowthComponent, HungerComponent>();
+        while (query.MoveNext(out var uid, out var slime, out var growthComp, out var hungerComp))
+        {
+            if (_mobState.IsDead(uid))
+                continue;
+
+            if (growthComp.CurrentStage == growthComp.Stages[0])
+                continue;
```

Comment:
Again that stage thing should be an enum.
```suggestion
            if (_mobState.IsDead(uid)
                || growthComp.CurrentStage == growthComp.Stages[0])
                continue;